File: server/Simulation/Components/BodyDamageComponent.cs
using server.ECS;
using server.Enums;

namespace server.Simulation.Components;

[Component(ComponentType = ComponentType.BodyDamage, NetworkSync = true)]
public struct BodyDamageComponent(float damage = 1)
{
    public long ChangedTick = NttWorld.Tick;
    public readonly float Damage = damage;
}

File: pixiejsClient/src/app/ecs/components/BodyDamageComponent.ts
import { Component, component, serverField } from "../core/Component";
import { ServerComponentType } from "../../enums/ComponentIds";
import { NTT } from "../core/NTT";

@component(ServerComponentType.BodyDamage)
export class BodyDamageComponent extends Component {
  // changedTick is inherited from Component base class
  @serverField(1, "f32") damage: number;

  constructor(ntt: NTT, damage: number = 1) {
    super(ntt);
    this.damage = damage;
  }
}
